payment.php

```diff
@@ -24,158 +24,58 @@
 /*
  * SET POST PARAMETERS TO VARIABLES AND CHECK IF THEY EXIST
  */
-$uuid = htmlspecialchars(isset($_REQUEST['uuid']) ? $_REQUEST['uuid'] : "");
 $get_order = htmlspecialchars(isset($_REQUEST['get_order']) ? $_REQUEST['get_order'] : "");
 $finish_order = htmlspecialchars(isset($_REQUEST['finish_order']) ? $_REQUEST['finish_order'] : "");
 
-$fiat_amount = htmlspecialchars($_POST['price']);
-$currency = htmlspecialchars($_POST['currency']);
-$order_id = htmlspecialchars($_POST['order_id']);
+$order_hash = htmlspecialchars(isset($_REQUEST['order']) ? $_REQUEST['order'] : "");
 
 $system_url = $blockonomics->getSystemUrl();
 $ca->assign('system_url', $system_url);
 
-/**
- * Try to generate a new Bitcoin address for payment.
- * Set error for template if not able
- * Return BTC address if success, null if error
- */
-function generate_address($blockonomics, $ca) {
-	$response_obj = $blockonomics->getNewBitcoinAddress();
-	$error = $blockonomics->checkForErrors($response_obj);
-	if ($error) {
-		$error = True;
-		$ca->assign('error', $error);
-		return null;
+if($get_order){
+	$blockonomics_currency = htmlspecialchars(isset($_REQUEST['blockonomics_currency']) ? $_REQUEST['blockonomics_currency'] : "");
+	$existing_order = $blockonomics->processOrderHash($get_order, $blockonomics_currency);
+	// No order exists, exit
+	if(is_null($existing_order->id_order)) {
+		exit;
 	} else {
-		$btc_address = $response_obj->address;
-		$ca->assign('btc_address', $btc_address);
-		return $btc_address;
+		header("Content-Type: application/json");
+		exit(json_encode($existing_order));
 	}
+}else if($finish_order){
+	$existing_order = $blockonomics->getOrderByAddress($finish_order);
```

Comment:
finish_order should directly take order_id as parameter. Then we can delete getOrderByAddress function also

Reply:
Updated in a74c4e5

Reply:
Please remove *getOrderByAddress* as it is now unused

Reply:
Function is still used in the callback file:
https://github.com/blockonomics/whmcs-bitcoin-plugin/blob/master/modules/gateways/callback/blockonomics.php#L44